contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -882,10 +888,149 @@ class tsqlCommonMutator : public TSqlParserBaseListener
 			rewritten_query_fragment.emplace(std::make_pair(ctx->id().back()->start->getStartIndex(), std::make_pair(::getFullText(ctx->id().back()), delimit_identifier(ctx->id().back()))));
 	}
 
+	void exitFunction_call(TSqlParser::Function_callContext *ctx) override
+	{
+		/* Handles rewrite of geospatial function calls but inside body of CREATE/ALTER View, Procedure, Function */
+		if (ctx->spatial_proc_name_server_database_schema())
+		{
+			if (ctx->spatial_proc_name_server_database_schema()->schema) throw PGErrorWrapperException(ERROR, ERRCODE_FEATURE_NOT_SUPPORTED, "Remote procedure/function reference with 4-part object name is not currently supported in Babelfish", getLineAndPos(ctx));
```

Comment:
are we testing this error message anywhere?

Reply:
yes, we have this test in our verify file
```
DECLARE @referencePoint geometry = geometry::Point(0.0, 0.0, 4326);
SELECT dbo.YourTable.PointColumn.STDistance(@referencePoint) AS Distance FROM YourTable ORDER BY PointColumn.STX;
GO
```